packages/react/dialog/src/Dialog.tsx

```diff
@@ -240,7 +240,10 @@ const DialogContentImpl = React.forwardRef((props, forwardedRef) => {
             // (closed !== unmounted when animating out)
             trapped={context.open}
             onMountAutoFocus={onOpenAutoFocus}
-            onUnmountAutoFocus={onCloseAutoFocus}
+            onUnmountAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
```

Comment:
This one isn't strictly necessary right now is it?

Reply:
Without it, the same bug could occur in dialog right?

Reply:
I guess technically it may happen with Safari because it won't focus the button?

Reply:
It will be an issue in any browser AFAIK...

**Without this change**

https://user-images.githubusercontent.com/175330/123978222-98c19c80-d9b7-11eb-8504-9ad080b77c21.mp4


**With this change**

https://user-images.githubusercontent.com/175330/123978038-6fa10c00-d9b7-11eb-8eb6-1c416c54dc97.mp4

Reply:
Why though? Did we remove the fallback to previous `activeElement`?

Reply:
I have an input inside the dialog with an `autoFocus` attribute (to match the original bug report) so:
- when it opens, the previous `activeElement` will be the `input` at the time the `FocusScope` effect runs
- when it closes it will focus the `input` and not the trigger which gets unmounted and we lose focus

Reply:
Oh wow, that's super unexpected that the autofocus's input steals it earlier. I think I saw Diego mention something like that on twitter, and that it's potentially a React bug.

Reply:
https://twitter.com/diegohaz/status/1408180632933388289

Reply:
I'm not sure if it is a React bug. If you think about it, React will render before it runs the effects. The render will mean the browser will `autoFocus` the input, then the effect will run and get that as the `activeElement`.

Reply:
Yeah I guess the browser can do it quicker than React.